### PR TITLE
v8 Spec

### LINKED ISSUE #700: Support for ImageSource, similar to VideoSource
I would appreciate support for an image overlay source, similar to the video source but for images. 

Would a naïve porting of the video source code suffice for this?

Would it be complex to have support for changing the coordinates on the source after it has been added (goes for both video and image)?

### PR DESCRIPTION
**Pull request by tmcw**
Implements breaking changes of https://github.com/mapbox/mapbox-gl-style-spec/pull/264

* [x] Rename symbol-min-distance to symbol-spacing. 
* [x] Flip coordinate order of video coordinates
* [x] Support font stacks as arrays
* [x] background-image ⇢ background-pattern
* [x] line-image ⇢ line-pattern
* [x] fill-image ⇢ fill-pattern
* [x] Rename video url to urls.
* [x] ImageSource https://github.com/mapbox/mapbox-gl-js/issues/700 Fixes #700
* [ ] Circle renderer https://github.com/mapbox/mapbox-gl-style-spec/issues/276

### CODE REVIEW

File: js/style/style.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ var browser = require('../util/browser');
 var Dispatcher = require('../util/dispatcher');
 var Point = require('point-geometry');
 var AnimationLoop = require('./animation_loop');
-var validate = require('mapbox-gl-style-lint/lib/validate/latest');
+var validate = require('mapbox-gl-style-spec/lib/validate/latest');
```
Comment (mourner) [inline js/style/style.js]: wat

Comment (tmcw) [inline js/style/style.js]: See the bullet point, relies on https://github.com/mapbox/mapbox-gl-style-spec/pull/259

**Comment by tmcw** [comment]
Moved to https://github.com/mapbox/mapbox-gl-js/pull/1192

**Comment by Yuki-Zhao** [comment]
Why is the name changed to line-pattern, fill-pattern?
the point is icon-image

**Pull request closed without merging**